Microsoft.Toolkit.Uwp.UI.Controls/RemoteDevicePicker/RemoteDevicePicker.cs

```diff
@@ -0,0 +1,358 @@
+            }
+
+            var authType = typeof(RemoteSystemAuthorizationKind).GetEnumNames().OrderBy(a => a.ToString()).ToList();
+            _authorizationType.ItemsSource = authType;
+
+            if (_authorizationKindFilter != null)
+            {
+                _authorizationType.SelectedValue = _authorizationKindFilter.RemoteSystemAuthorizationKind.ToString();
+            }
+        }
+
+        private void BuildFilters(RemoteSystemDiscoveryTypeFilter discoveryFilter, RemoteSystemAuthorizationKindFilter authorizationKindFilter, RemoteSystemStatusTypeFilter statusFilter)
+        {
+            var filters = new List<IRemoteSystemFilter>();
+            if (discoveryFilter != null)
+            {
+                filters.Add(discoveryFilter);
+            }
+
+            if (authorizationKindFilter != null)
+            {
+                filters.Add(authorizationKindFilter);
+            }
+
+            if (statusFilter != null)
+            {
+                filters.Add(statusFilter);
+            }
+
+            var remoteDeviceHelper = new RemoteDeviceHelper(filters);
+            RemoteSystems = remoteDeviceHelper.RemoteSystems;
+
+            UpdateProgressRing(true);
+            UpdateList();
+        }
+
+        private void ListDevices_ContainerContentChanging(ListViewBase sender, ContainerContentChangingEventArgs args)
+        {
+            var model = (RemoteSystem)args.Item;
+            switch (model.Status)
+            {
+                case RemoteSystemStatus.Available:
+                    args.ItemContainer.IsEnabled = true;
+                    break;
+
+                case RemoteSystemStatus.DiscoveringAvailability:
+                case RemoteSystemStatus.Unavailable:
+                case RemoteSystemStatus.Unknown:
+                    args.ItemContainer.IsEnabled = false;
+                    break;
+            }
+        }
+
+        private void ListDevices_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (_listDevices.SelectionMode == ListViewSelectionMode.Single)
+            {
+                ReturnDevices();
+                Hide();
+            }
+            else
+            {
+                if (_listDevices.SelectedItems.Count > 0)
+                {
+                    IsSecondaryButtonEnabled = true;
+                }
+                else
+                {
+                    IsSecondaryButtonEnabled = false;
+                }
+            }
+        }
+
+        private void ListDevices_DoubleTapped(object sender, DoubleTappedRoutedEventArgs e)
+        {
+            ReturnDevices();
+            Hide();
+        }
+
+        private void ListDeviceTypes_SelectionChanged(object sender, SelectionChangedEventArgs e) => UpdateList();
+
+        private void UpdateList()
+        {
+            var bindingList = new ObservableCollection<RemoteSystem>();
```

Comment:
Rome API has `RemoteSystemWatcher_RemoteSystemAdded` and `RemoteSystemWatcher_RemoteSystemUpdated`. And both the events are fired when the watcher is initialized. That is the reason why i see the quick flicker in list load/reload.

Reply:
Would it be better to have a copy of the remote system list, so when we get an update we can update the copy in place so that it doesn't disrupt the existing view?